lib/active_model/serializer/reflection.rb

```diff
@@ -75,10 +75,10 @@ def value(serializer)
 
         if block
           block_value = instance_exec(serializer, &block)
-          if block_value == :nil
-            serializer.read_attribute_for_serialization(name)
-          else
+          if block_value != :nil
             block_value
+          elsif @_include_data
```

Comment:
I'm not sure about the question but it is set to `true` by default so includes the relationship `data` by default. The point of the PR is to prevent calling the relationship if `include_data` is set to `false`.